### PR TITLE
break(flake): use upstream options...

### PR DESCRIPTION
**Pull request by phaer**
* use nixpkgs.flake settings to add nixpkgs
* drop symlink flake
* replace it with registerSelf, which uses the flake registry
* don't add all inputs of `self` anymore, they can quickly be referenced with `self.inputs` anyway.

### CODE REVIEW

**Comment by phaer** [comment]
I did no lib.mkRenamedOptionModule for symlinkFlake -> registerSelf, as it's not just a rename.  

We say "stable" in the README, but we don't really have releases or anything to communicate 
breaking changes yet, I think? Technically, there's another breaking change as this doesn't add all inputs of `self` anymore - on the main branch for a few days.

**Comment by zimbatm** [comment]
Ok. I assumed because you were duplicating some of the stuff in https://github.com/nix-community/infra/pull/1436

**Review by zimbatm — APPROVED** [review]

File: nixos/common/flake.nix [hunk 1]
```diff
@@ -55,14 +40,20 @@ in
...
-    nix.settings.flake-registry = lib.mkDefault flake-registry;
 
-    system.extraSystemBuilderCmds = lib.optionalString cfg.symlinkFlake ''
-      ln -s ${cfg.flake} $out/flake
-    '';
+    nixpkgs.flake = {
+      source = lib.mkDefault (cfg.flake.inputs.nixpkgs or null);
+    };
+
+    nix.registry = lib.optionalAttrs cfg.registerSelf {
```
Comment (Mic92) [inline nixos/common/flake.nix:51]: I think we can actually enable this for the desktop profile?

**Pull request merged**